Zadanie53

jtermi opened the pull request.

jtermi committed:
zadanie zrobione

jtermi committed:
zadanie zrobione

slawekludwiczak commented on src/main/java/pl/javastart/task/engine/Category.java:
```diff
@@ -0,0 +1,24 @@
+        double result;
+
+        switch (product.getCategory()) {
+            case "Spożywcze", "Dziecięce":
+                result = product.getPrice() * 1.07;
+                System.out.println("Stawka VAT wynosi 7%");
+                break;
+            case "Biurowe":
+                result = product.getPrice() * 1.14;
+                System.out.println("Stawka VAT wynosi 14%");
+                break;
+            default:
+                result = product.getPrice() * 1.23;
+                System.out.println("Stawka VAT wynosi 23%");
```
W metodach, które obliczają i zwracają wynik nie dodawaj System.out.printlnów. Staraj się rozdzielać logike programu, żeby logika aplikacji i interakcja z użytkownikiem (System.outy, Scanner) były w osobnych klasach/metodach

slawekludwiczak commented on src/main/java/pl/javastart/task/engine/Product.java:
```diff
@@ -0,0 +1,56 @@
+        this.category = category;
+    }
+
+    public void productInformation() {
+        System.out.printf("Nazwa produktu: %s\n", name);
+        System.out.printf("Opis Produktu: %s\n", description);
+        System.out.printf("Cena netto: %.2f PLN\n", price);
+        System.out.printf("Produkt należy do kategorii: %s \n", category);
+    }
```
To też generalnie jest ok, ale lepiej by było tutaj zwrócić Stringa z opisem, a wyświetlaniem zająć się już w main na zasadzie `System.out.println(product1.getInformation())`

jtermi committed:
Poprawione błędy:
- usunąłem out.println'y z metod logiki programu
- utworzylem osobne metody zwracające stringi.

Napotkałem problem:
- metoda obliczająca cenę brutto popełnia z jakiegoś powodu błąd wynik różni się od rzecziwistej wartośći o małą wartość - kilka znaków po przecinku. Nie wiem jak rozwiązać ten problem. Internet o tym otwarcie nie mówi :)
